doc/cookbook/namedRoutes/NamedRoutes.lhs

```diff
@@ -0,0 +1,314 @@
+# NamedRoutes - Using records to define APIs
+
+Servant offers a very natural way of constructing APIs with nested records, called `NamedRoutes`.
+
+This cookbook explains how to implement such nested-record-APIs using `NamedRoutes` through the example of a Movie Catalog.
+
+First, we start by constructing the domain types of our Movie Catalog.
+After, we show you how to implement the API type with the NamedRoutes records.
+Lastly, we make a Server and a Client out of the API type.
+
+However, it should be understood that this cookbook does _not_ dwell on the built-in servant combinators as the [<TODO title> cookbook ](<TODO url>) already covers that angle.
+
+## Why should we use `NamedRoutes` over the alternative `<:|>` operator?
+
+With `NamedRoutes`, we don’t need to care about the declaration order of the endpoints.
+For example, with the `<:|>` operator there’s room for error when the order of the API type
+
+```haskell,ignore
+type API1  =   "version" :> Get '[JSON] Version
+                <:|>  "movies" :> Get '[JSON] [Movie]
+```
+
+does not follow the `Handler` implementation order
+
+```haskell,ignore
+apiHandler :: (AsServerT Handler)
+apiHandler =   getMovies
+                 <:|> getVersion
+```
```

Comment:
Maybe for illustration add the annoying error message that you'd get in this case?

Reply:
I totally agree with that :)